catalog/core/catalog-core-commands/src/test/java/org/codice/ddf/commands/catalog/ValidateCommandTest.java

```diff
@@ -0,0 +1,149 @@
+        validators.add(mockValidator.getValidator());
+        assertThat(execute(validators), containsString("sample warning"));
+    }
+
+    @Test
+    public void testWithError() throws Exception {
+        List<MetacardValidator> validators = new ArrayList<>();
+        MockValidator mockValidator = new MockValidator();
+        mockValidator.errors = new ArrayList<>(Arrays.asList("sample error"));
+        validators.add(mockValidator.getValidator());
+        assertThat(execute(validators), containsString("sample error"));
+    }
+
+    @Test
+    public void testWithMultipleValidators() throws Exception {
+        List<MetacardValidator> validators = new ArrayList<>();
+        MockValidator mockValidator1 = new MockValidator();
+        MockValidator mockValidator2 = new MockValidator();
+        mockValidator1.id = "validator 1";
+        mockValidator2.id = "validator 2";
+        validators.add(mockValidator1.getValidator());
+        validators.add(mockValidator2.getValidator());
+        assertThat(execute(validators), containsString("validator 1"));
+        assertThat(execute(validators), containsString("validator 2"));
+    }
+
+    @Test
+    public void testWithErrorsAndWarnings() throws Exception {
+        List<MetacardValidator> validators = new ArrayList<>();
+        MockValidator mockValidator = new MockValidator();
+        mockValidator.errors = new ArrayList<>(Arrays.asList("error 1", "error 2"));
+        mockValidator.warnings = new ArrayList<>(Arrays.asList("warning 1", "warning 2"));
+        validators.add(mockValidator.getValidator());
+        assertThat(execute(validators), containsString("error 1"));
+        assertThat(execute(validators), containsString("error 2"));
+        assertThat(execute(validators), containsString("warning 1"));
+        assertThat(execute(validators), containsString("warning 2"));
+    }
+
+    @Test
+    public void testWithValidatorsEqualNull() throws Exception {
+        assertThat(execute(null), containsString("No validators have been configured"));
+    }
+
+    @Test
+    public void testWithValidatorsSizeZero() throws Exception {
+        assertThat(execute(new ArrayList<>()), containsString("No validators have been configured"));
+    }
+
+    private String execute(List<MetacardValidator> validators) throws Exception {
+        return execute(validators, "empty.xml");
+    }
+
+    private String execute(List<MetacardValidator> validators, String filename) throws Exception {
+        ConsoleOutput consoleOutput = new ConsoleOutput();
+        consoleOutput.interceptSystemOut();
+
+        ValidateCommand command = new ValidateCommand();
+        command.filename = Paths.get(testFolder.getRoot().getAbsolutePath(), filename).toString();
+        command.validators = validators;
+        command.execute();
+        consoleOutput.resetSystemOut();
+        return consoleOutput.getOutput();
+    }
+
+    private class MockValidator {
```

Comment:
It doesn't always throw an exception, it just throws an exception when whatever it is validating has errors or warnings. I'd stick with the name "mockValidator" for this reason.